subbasins.py

```diff
@@ -101,6 +102,9 @@ def delineate(input_csv: str, output_prefix: str):
     # Read the outlet points CSV file and put the data into a Pandas DataFrame
     # (I call the outlet points gages, because I usually in delineated watersheds at streamflow gages)
     gages_gdf = make_gages_gdf(input_csv)
+    # TODO: add some amount of buffer?
+    all_points = unary_union(gages_gdf['geometry']).buffer(1)
```

Comment:
this is what this looks like w the test data 
![image](https://github.com/user-attachments/assets/3597c171-2050-42f2-8cf8-a11495977b48)